src/basic/sysctl-util.c

```diff
@@ -58,6 +82,14 @@ int sysctl_write(const char *property, const char *value) {
 
         log_debug("Setting '%s' to '%s'", p, value);
 
+        if (shadow) {
+                int r;
```

Comment:
we have this rule in our codebase that `int r` always is dfined at outermost function scope, because it's *the* variable to store function return codes in. Generally our rules to allocat evariables at minimal scope, just `int r` gets that one exception.